tests/test_for_support/test_for_matrix.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017 Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Ensure the expected functioning of ``memote.support.matrix``."""
+
+from __future__ import absolute_import
+
+import cobra
+import pytest
+from cobra.exceptions import Infeasible
+
+import memote.support.matrix as matrix
+from memote.utils import register_with
+
+MODEL_REGISTRY = dict()
+
+
+@register_with(MODEL_REGISTRY)
+def three_components_closed(base):
```

Comment:
I don't quite understand why you make a distinction between 3 and 4 component models?

Considering they yield identical results in many of the tests below.

Reply:
The only difference between the two was that the 4-compartment system has an additional, irreversible reaction, while the 3-comp system only has reversible reactions. However given that the results don't really differ, it can be taken out if needed.